File: flutter-app/lib/models/salon.dart
class Salon {
  final int? id;
  final String? salonName;
  final String? address;
  final String? city;
  final String? state;
  final String? pincode;
  final String? contactNumber;
  final String? imageUrl;
  final String? approvalStatus;
  final String? openTime;
  final String? closeTime;
  final int? ownerId;
  final String? category;
  final bool? isLive;
  final double? latitude;
  final double? longitude;
  final String? establishedYear;
  final String? description;
  final int? numberOfStaff;
  final String? specialities;
  final String? languages;
  final String? serviceArea;

  Salon({
    this.id,
    this.salonName,
    this.address,
    this.city,
    this.state,
    this.pincode,
    this.contactNumber,
    this.imageUrl,
    this.approvalStatus,
    this.openTime,
    this.closeTime,
    this.ownerId,
    this.category,
    this.isLive,
    this.latitude,
    this.longitude,
    this.establishedYear,
    this.description,
    this.numberOfStaff,
    this.specialities,
    this.languages,
    this.serviceArea,
  });

  factory Salon.fromJson(Map<String, dynamic> json) {
    return Salon(
      id: json['id'] as int?,
      salonName: json['salonName'] as String?,
      address: json['address'] as String?,
      city: json['city'] as String?,
      state: json['state'] as String?,
      pincode: json['pincode'] as String?,
      contactNumber: json['contactNumber'] as String?,
      imageUrl: json['imageUrl'] as String?,
      approvalStatus: json['approvalStatus'] as String?,
      openTime: json['openTime'] as String?,
      closeTime: json['closeTime'] as String?,
      ownerId: json['ownerId'] as int?,
      category: json['category'] as String?,
      isLive: json['isLive'] as bool?,
      latitude: (json['latitude'] as num?)?.toDouble(),
      longitude: (json['longitude'] as num?)?.toDouble(),
      establishedYear: json['establishedYear'] as String?,
      description: json['description'] as String?,
      numberOfStaff: json['numberOfStaff'] as int?,
      specialities: json['specialities'] as String?,
      languages: json['languages'] as String?,
      serviceArea: json['serviceArea'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'salonName': salonName,
      'address': address,
      'city': city,
      'state': state,
      'pincode': pincode,
      'contactNumber': contactNumber,
      'imageUrl': imageUrl,
      'approvalStatus': approvalStatus,
      'openTime': openTime,
      'closeTime': closeTime,
      'ownerId': ownerId,
      'category': category,
      'isLive': isLive,
      'latitude': latitude,
      'longitude': longitude,
      'establishedYear': establishedYear,
      'description': description,
      'numberOfStaff': numberOfStaff,
      'specialities': specialities,
      'languages': languages,
      'serviceArea': serviceArea,
    };
  }
}

File: backend/src/main/java/com/salonvala/salonmanagement/entity/Salon.java
package com.salonvala.salonmanagement.entity;

import com.salonvala.salonmanagement.entity.User;
import com.salonvala.salonmanagement.enums.SalonApprovalStatus;
import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "salons")
public class Salon {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String salonName;

    private String address;

    private String city;

    private String state;

    private String pincode;

    private String contactNumber;

    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String imageUrl;

    @Enumerated(EnumType.STRING)
    private SalonApprovalStatus approvalStatus = SalonApprovalStatus.PENDING;

    @ManyToOne
    @JoinColumn(name = "owner_id")
    private User owner;

    private String openTime;

    private String closeTime;

    private Boolean isLive = false;

    // Geographic location (for maps / directions)
    private Double latitude;

    private Double longitude;

    // Salon category (e.g., "Men's Salon", "Women's Salon", "Unisex", "Makeup", "Facial", "Hair Care")
    private String category;

    // Additional business details
    private String establishedYear;

    @Column(length = 2000)
    private String description;

    private Integer numberOfStaff;

    private String specialities;

    private String languages;

    // Service coverage area description
    private String serviceArea;
    
    // Landmark for location
    private String landmark;
    
    // Home service availability
    private Boolean homeServiceAvailable = false;
    
    // Working days (e.g., "Mon,Tue,Wed,Thu,Fri,Sat")
    private String workingDays;
    
    // Rejection reason (when rejected by admin)
    @Column(length = 1000)
    private String rejectionReason;
    
    // ==================== OWNER DETAILS ====================
    private String ownerName;
    private String ownerPhone;
    private String ownerEmail;
    private String ownerAge;
    private String ownerGender;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String ownerProfileImageUrl;
    
    // ==================== DOCUMENTS ====================
    // Aadhaar
    private String aadhaarNumber;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String aadhaarFrontImageUrl;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String aadhaarBackImageUrl;
    
    // PAN Card
    private String panNumber;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String panCardImageUrl;
    
    // Shop License
    private String shopLicenseNumber;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String shopLicenseImageUrl;
    
    // GST
    private String gstNumber;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String gstCertificateImageUrl;
    
    // ==================== PHOTOS ====================
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String liveSelfieImageUrl;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String shopFrontImageUrl;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String shopInsideImage1Url;
    
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String shopInsideImage2Url;

    // Timestamp when salon was registered
    @CreationTimestamp
    @Column(updatable = false)
    private LocalDateTime createdAt;

    // Timestamp when salon was reviewed by admin
    private LocalDateTime reviewedAt;

    // Admin who reviewed the salon
    private Long reviewedBy;
}
